platform/darwin/test/MGLPredicateTests.mm

```diff
@@ -576,6 +576,66 @@ - (void)testComparisonExpressionArray {
         NSArray *expected = @[@"==", @1, @2];
         XCTAssertEqualObjects([NSPredicate predicateWithFormat:@"1 = 2"].mgl_jsonExpressionObject, expected);
     }
+    {
+        NSPredicate *predicate = [NSPredicate predicateWithFormat:@"x == YES"];
+        NSArray *jsonExpression = @[@"==", @[@"get", @"x"], @YES];
+        XCTAssertEqualObjects(predicate.mgl_jsonExpressionObject, jsonExpression);
```

Comment:
Also test that reversing the order of the operands produces the same kind of expression.